### PR TITLE
Bump @nextcloud/vue to 3.1.0

### PR DESCRIPTION
**Pull request by PVince81**
This brings in:
- the new edit component (which we can replace later in https://github.com/nextcloud/spreed/pull/4333)
- Scrollable app sidebar (which is the reason for major bump)
- Avatar caching.
- Other small things

### Todos

- [x] cleanup sidebar css hacks
- [x] ~~potentially blocked by https://github.com/nextcloud/spreed/pull/4195 (issues mentioned there)~~ => https://github.com/nextcloud/spreed/pull/4473#issuecomment-717083734

### Tests

I did a quick test to see if all still works:
- [x] scrolling in left sidebar still works fine
- [x] scrolling in participant list works fine
- [x] sending (typing)/receiving messages still works, with emojis
- [x] remove hacks
- [x] check settings panel for regression
- [x] verify chat scrolling with lots of messages when conversation in sidebar
- [x] verify regular chat scrolling

Obsoletes https://github.com/nextcloud/spreed/pull/4457

### CODE REVIEW

**Review by skjnldsv — COMMENTED** [review]
please also cleanup old css hacks
https://github.com/nextcloud/spreed/blob/fc1cbfce48c2cb3a4ca6a12bce422ee6f118a10b/src/components/RightSidebar/RightSidebar.vue#L273-L286

**Comment by PVince81** [comment]
I had a quick look at https://github.com/nextcloud/spreed/pull/4195 and I don't see any difference before and after the Vue update, so I'd say it's not a blocker. The PR in question needs further work and might require further changes in nextcloud-vue, for which we can do future releases + version bumps.

**Comment by PVince81** [comment]
@skjnldsv when I remove the whole block you mentioned, there's a double scrollbar again.

I compared the sidebar before and after the vue update, removing or adding the hack back, I saw no difference in behavior.
So I'm not sure how the new tab scrollbar approach is supposed to help in this situation.

**Comment by skjnldsv** [comment]
> @skjnldsv when I remove the whole block you mentioned (specifically, the overflow:hidden), there's a double scrollbar again.

Yeah, it's because of https://github.com/nextcloud/spreed/blob/38d0870c18ddd1e54e0bc3cc48e03aeb2ca78b76/src/components/RightSidebar/RightSidebar.vue#L268-L270
You need to remove this as well.
Please do not remove the outline! This is an accessibility feature and is here on purpose.
Generally speaking you should not override components styling. If it's here, it's here on purpose and for all apps.

**Comment by PVince81** [comment]
Indeed. so this will reopen https://github.com/nextcloud/spreed/issues/2946 and we need to find a different solution for it, maybe rework the tab contents layout. Ideally the scroll container should start below the field instead of having the field floating and overlapping. This means having a scroll container even deeper inside the tab content and not using the tab content's own scroller.

**Comment by skjnldsv** [comment]
> Ideally the scroll container should start below the field instead of having the field floating and overlapping.

No need, just put the padding on the form, and it will have the background under it?
Position sticky is exactly why it was designed for :)

**Comment by skjnldsv** [comment]
Also works perfectly fine for my by defaults

**Comment by PVince81** [comment]
@skjnldsv easier said than done. I'll need some CSS (or even Vue) trickery to remove the padding on the parent ".app-sidebar-tabs__content" to add it back to the input field. But only do that when the participants tab is selected.

**Comment by PVince81** [comment]
never mind, was looking at the wrong element...

File: src/components/RightSidebar/RightSidebar.vue [hunk 1]
```diff
@@ -260,28 +260,11 @@ export default {
...
-	outline: none; /* remove the weird border that appears on focus */
-}
-
-/* Force scroll bars in tabs content instead of in whole sidebar. */
-::v-deep .app-sidebar-tabs {
-	height: calc(100% - 80px);
-
-	&__content {
-		overflow: hidden;
+	padding-top: 0px;
```
Comment (skjnldsv) [inline src/components/RightSidebar/RightSidebar.vue:264]: I still don't see why that is necessary :thinking:

Comment (PVince81) [reply src/components/RightSidebar/RightSidebar.vue:264]: after my PR:
image

if I remove padding-top: 0
image

so I need to compensate by removing the 8px from the content (the top green border)
image

Comment (PVince81) [reply src/components/RightSidebar/RightSidebar.vue]: adjusted and squashed based on more hints from @skjnldsv, thanks!

**Comment by PVince81** [comment]
I've verified the scrolling with lots of messages in the files app sidebar and it worked fine! (had to use https://github.com/nextcloud/spreed/pull/4291 for testing)

If CI is green I think we can move forward with this PR.

**Review by skjnldsv — APPROVED** [review]

File: src/components/RightSidebar/RightSidebar.vue [hunk 2]
```diff
@@ -260,29 +261,14 @@ export default {
...
-	&__content {
-		overflow: hidden;
-
-		section {
-			height: 100%;
-
-			overflow-y: auto;
-		}
-	}
+::v-deep .app-navigation-search {
```
Comment (skjnldsv) [inline src/components/RightSidebar/RightSidebar.vue:268]: You most likely don't need v-deep as the root component SearchBox is used in the ParticipantTab component and will share the same vue-scope-id :)
https://github.com/nextcloud/spreed/blob/d0094567770ef799f039c6f57c1d33ca19f6f60e/src/components/RightSidebar/Participants/ParticipantsTab.vue#L275

You can move this there

Comment (PVince81) [reply src/components/RightSidebar/RightSidebar.vue]: adjusted and squashed

at that point I forgot that I had direct access and that it wasn't an external component

Comment (skjnldsv) [reply src/components/RightSidebar/RightSidebar.vue]: Yes, it gets confusing very fast :p

**Comment by PVince81** [comment]
I forgot to verify the sidebar chat during a call (I only checked in the files app).
It has a double scrollbar now, raised here https://github.com/nextcloud/spreed/issues/4492

**Pull request merged**